src/NexusMods.App.UI/Pages/LibraryPage/NexusModsModPageLibraryItemModel.cs

```diff
@@ -103,14 +110,42 @@ public NexusModsModPageLibraryItemModel(IObservable<IChangeSet<NexusModsLibraryI
                 {
                     var (numInstalled, numTotal, isExpanded) = tuple;
                     self.InstallButtonText.Value = ILibraryItemWithInstallAction.GetButtonText(numInstalled, numTotal, isExpanded);
+                    self.UpdateButtonText.Value = ILibraryItemWithUpdateAction.GetButtonText(self._numUpdatable, numTotal, isExpanded);
                 })
                 .AddTo(disposables);
         });
+        
+        var setVersionDisposable = ChildrenObservable
+            .ToCollection()
+            .Subscribe(items =>
+            {
+                // N20250116
+                // Note(sewer): Design says put highest version of child here.
+                //
+                // There is no 'standard' for version fields, as some sources
+                // like the Nexus website allow you to specify anything in the version field.
+                // We do a 'best effort' here by trying to parse as SemVer and using
+                // that. There are some possible alternatives, e.g. 'by upload date',
+                // however; that then requires additional logic, to not pick up other
+                // mods on the same mod page, etc. So we go for something simple for now.
+                var maxVersion = items
+                    .OfType<ILibraryItemWithVersion>()
+                    .Max(x => NuGetVersion.TryParse(x.Version.Value, out var version) 
+                        ? version : new NuGetVersion(0, 0, 0));
+
+                if (maxVersion != null)
+                {
+                    // SAFETY: Not null because >= 1 item.
+                    _preUpdateVersion = maxVersion!.ToString();
+                    Version.Value = _preUpdateVersion;
+                } 
+            });
```

Comment:
I think going to change `ChildrenObservable` to a private getter to prevent this. It's never supposed to be accessed by anything other than `TreeDataGridItemModel` itself.